pyclowder/connectors.py

```diff
@@ -391,16 +402,18 @@ def _process_message(self, body):
         if not host.endswith('/'): host += '/'
         secret_key = body.get('secretKey', '')
         retry_count = 0 if 'retry_count' not in body else body['retry_count']
-        resource = self._build_resource(body, host, secret_key)
+        clowder_version = int(body.get('clowderVersion', os.getenv('CLOWDER_VERSION', '1')))
+        resource = self._build_resource(body, host, secret_key, clowder_version)
         if not resource:
             logging.error("No resource found, this is bad.")
             return
 
         # register extractor
```

Comment:
I think we can get rid of the code that registers the extractor, and we should fully depend on the heartbeat.